InternetHospital.BusinessLogic/Services/PatientService.cs

```diff
@@ -24,6 +25,59 @@ public PatientService(ApplicationContext context, IFilesService uploadingFiles)
             _uploadingFiles = uploadingFiles;
         }
 
+        public async Task<PatientDetailedModel> Get(int id)
+        {
+            PatientDetailedModel returnedPatient = null;
+            var searchedPatient = await _context.Users.Include(p=> p.IllnessHistories).FirstOrDefaultAsync(pa=> pa.Id == id);
+            if (searchedPatient != null)
+            {  
+                returnedPatient = Mapper.Map<User, PatientDetailedModel>(searchedPatient);
+            }
+            return returnedPatient;
+        }
+        public (IEnumerable<IllnessHistoryModel> histories, int count) GetFilteredHistories(IllnessHistorySearchModel queryParameters)
+        {
+            var histories = _context.IllnessHistories.AsQueryable();
+            if (queryParameters.SearchFromDate != null)
+            {
+                var fromDate = Convert.ToDateTime(queryParameters.SearchFromDate);
+                histories = histories
+                    .Where(d => d.ConclusionTime >= fromDate);
+            }
+            if (queryParameters.SearchToDate != null)
+            {
+                var toDate = Convert.ToDateTime(queryParameters.SearchToDate);
+                toDate = toDate.AddDays(1);
```

Comment:
Necessary kostil?

Reply:
The day that returns from front is the midnight of the day, and I need to include this day for proper filtering